### PR TITLE
chore: simplify typing

### PR DESCRIPTION
**Pull request by septs**
closes #

### CODE REVIEW

File: packages/.eslintrc.json [hunk 1]
```diff
@@ -26,6 +26,7 @@
         "unused-imports/no-unused-imports-ts": "warn",
         "@typescript-eslint/await-thenable": "error",
         "@typescript-eslint/no-for-in-array": "error",
+        "@typescript-eslint/consistent-indexed-object-style": ["error", "record"],
```
Comment (septs) [inline packages/.eslintrc.json:29]: https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/consistent-indexed-object-style.md

CC @DimensionDev/dev-maskbook

File: packages/maskbook/src/social-network/types.ts [hunk 2]
```diff
@@ -241,15 +241,9 @@ export namespace SocialNetworkUI {
             classes?: () => Props extends withClasses<infer T> ? Partial<Record<T, string>> : never
             props?: (props: Props) => Props
         }
-        export interface I18NOverwrite {
-            [namespace: string]: I18NOverwriteNamespace
```
Comment (Jack-Works) [inline packages/maskbook/src/social-network/types.ts:245]: The key here has its meaning. It is used like "comment" to this type.

File: packages/maskbook/src/plugins/Wallet/apis/coingecko.ts [hunk 3]
```diff
@@ -3,13 +3,9 @@ import { currentEtherPriceSettings } from '../settings'
 
 const ETH_PRICE_POLLING_DELAY = 30 /* seconds */ * 1000 /* milliseconds */
 
-interface PriceRecord {
-    [currency: string]: number
```
Comment (Jack-Works) [inline packages/maskbook/src/plugins/Wallet/apis/coingecko.ts:7]: The key here has its meaning. It is used like "comment" to this type.

File: packages/maskbook/src/settings/createSettings.tsx [hunk 4]
```diff
@@ -94,9 +94,7 @@ export function createGlobalSettings<T extends browser.storage.StorageValue>(
     return settings
 }
 
-export interface NetworkSettings<T> {
-    [networkKey: string]: ValueRef<T> & { ready: boolean; readyPromise: Promise<T> }
```
Comment (Jack-Works) [inline packages/maskbook/src/settings/createSettings.tsx:98]: The key here has its meaning. It is used like "comment" to this type.

File: packages/.eslintrc.json [hunk 5]
```diff
@@ -23,6 +23,7 @@
         "lodash/import-scope": ["error", "member"],
         "unused-imports/no-unused-imports-ts": "warn",
         "@typescript-eslint/await-thenable": "error",
-        "@typescript-eslint/no-for-in-array": "error"
+        "@typescript-eslint/no-for-in-array": "error",
+        "@typescript-eslint/consistent-indexed-object-style": ["error", "record"]
```
Comment (Jack-Works) [inline packages/.eslintrc.json:27]: Should we really prefer this one? It seems like we have many cases that the indexed key `{ [chainID: string]: T }` has its meaning of documenting/commenting the code meanwhile the `Record<string, T>` cannot represent.

Comment (septs) [reply packages/.eslintrc.json:27]: may need `@masknet/types`, provide shared type support system

**Review by zhouhanseng — DISMISSED** [review]

File: packages/.eslintrc.json (hunk 5 above)
Comment (Jack-Works) [reply packages/.eslintrc.json:27]: I mean, in the only case of `{ [key: string]: T }`, that `key` is a meaningless name, the choice `Record<string, T>` is good. In other cases, those names are used as the comment to the type.

**Comment by septs** [comment]
@Jack-Works why

**Comment by Jack-Works** [comment]
@septs I think we should discuss this rule in detail.

File: packages/.eslintrc.json (hunk 5 above)
Comment (guanbinrui) [reply packages/.eslintrc.json:27]: Agree with @Jack-Works. Both styles are acceptable.

**Comment by septs** [comment]
@Jack-Works require disable the rule, but other approve

**Review by guanbinrui — APPROVED** [review]

**Review by zhouhanseng — APPROVED** [review]

**Pull request merged**